### PR TITLE
Finish Chapter 3

### PR DESCRIPTION
**Pull request by heimrih**
ready

### CODE REVIEW

**Comment by heimrih** [comment]
ready

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvu3010) [inline app/assets/stylesheets/static_pages.scss]: remove empty file you have created

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
```
Comment (ngocvu3010) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
```

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,6 @@
...
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (ngocvu3010) [inline app/helpers/application_helper.rb]: please use i18n for text "Ruby on Rails Tutorial Sample App"
https://guides.rubyonrails.org/i18n.html

File: app/helpers/static_pages_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (ngocvu3010) [inline app/helpers/static_pages_helper.rb]: remove empty file you have created

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title(yield(:title)) %> </title>
```
Comment (ngocvu3010) [inline app/views/layouts/application.html.erb]:
```suggestion
    <title><%= full_title yield(:title) %> </title>
```

File: app/views/static_pages/about.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+
+<% provide :title, "About" %>
+<h1>About</h1>
+<p>
+The <a href="https://www.railstutorial.org/"><em>Ruby on Rails
```
Comment (ngocvu3010) [inline app/views/static_pages/about.html.erb]: using i18n for text
change"a href" by using link_to helper method

File: app/controllers/static_pages_controller.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class StaticPagesController < ApplicationController
+  def home
```
Comment (manhnt-0960) [inline app/controllers/static_pages_controller.rb]:
```suggestion
  def home; end
```

Comment (heimrih) [reply app/controllers/static_pages_controller.rb]: Do i repeat this for help and about too?

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (heimrih) [reply app/helpers/application_helper.rb]: Sorry i have read I am not sure how do i use it

File: app/views/static_pages/about.html.erb (hunk 6 above)
Comment (heimrih) [reply app/views/static_pages/about.html.erb]: image
 like this?

**Comment by heimrih** [comment]
ready

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -1,2 +1,14 @@
...
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
+
+  def default_url_options
+    { locale: I18n.locale }
+  end
+
```
Comment (manhnt-0960) [inline app/controllers/application_controller.rb]: remove empty line

File: app/controllers/static_pages_controller.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class StaticPagesController < ApplicationController
+  def home; end
+
+  def help; end
+
+  def about; end
+
```
Comment (manhnt-0960) [inline app/controllers/static_pages_controller.rb]: remove empty line

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (manhnt-0960) [reply app/helpers/application_helper.rb]: you should do follow this
 https://lokalise.com/blog/rails-i18n/

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

File: app/views/static_pages/help.html.erb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<%= content_for :title, t(".title") %>
+<title><%= t ".title" %></title>
+<h1><%= t "static_pages.help.title" %></h1>
+<p>
+
+Get help on the Ruby on Rails Tutorial at the
+
+<%= link_to "Rails Tutorial Help page", "https://www.railstutorial.org/help" %>.
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: "Rails Tutorial Help page => use i18n
need change all text in html.erb to use i18n

File: app/views/static_pages/help.html.erb [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+To get help on this sample app, see the
+<%= link_to "Ruby on Rails Tutorial book", "https://www.railstutorial.org/book" %>.
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: use i18n

File: config/application.rb [hunk 12]
```diff
@@ -14,6 +14,9 @@ class Application < Rails::Application
     # Settings in config/environments/* take precedence over those specified here.
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
-    # the framework and any gems in your application.
+    # the framework and any gems in your application. 
+    config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}')]
+config.i18n.available_locales = [:en, :vi]
```
Comment (ngocvt-0484) [inline config/application.rb]:
```suggestion
     config.i18n.available_locales = [:en, :vi]
```

File: config/routes.rb [hunk 13]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  scope "(:locale)", locale: /en|vi/ do
+    get '/about', to: 'static_pages#about', as: :about
+    get '/help', to: 'static_pages#help', as: :help
+    root 'static_pages#home'
+  end
+
 end
+
```
Comment (ngocvt-0484) [inline config/routes.rb:8]: remove empty line

File: config/routes.rb [hunk 14]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  scope "(:locale)", locale: /en|vi/ do
+    get '/about', to: 'static_pages#about', as: :about
```
Comment (ngocvt-0484) [inline config/routes.rb]: use " instead of '

**Comment by vanvtt-0952** [comment]
commented

**Comment by heimrih** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**